examples/shader/shadertree.py

```diff
@@ -109,17 +103,18 @@ class ShaderWidget(FloatLayout):
     def __init__(self, **kwargs):
         # Instead of using canvas, we will use a RenderContext,
         # and change the default shader used.
-        self.canvas = RenderContext(use_parent_projection=True)
+        self.canvas = RenderContext(use_parent_projection=True,
+                                    use_parent_modelview=True,
+                                    use_parent_frag_modelview=True)
 
-        # We create a framebuffer at the size of the window
-        # FIXME: this should be created at the size of the widget
         with self.canvas:
-            self.fbo = Fbo(size=Window.size, use_parent_projection=True)
+            self.fbo = Fbo(size=self.size)
+            self.fbo_color = Color(1, 1, 1, 1)
+            self.fbo_rect = Rectangle()
 
-        # Set the fbo background to black.
         with self.fbo:
-            Color(0, 0, 0)
-            Rectangle(size=Window.size)
+            ClearColor(0, 0, 0, 0)
```

Comment:
Hehehe, `Color()` + `Rectangle()` was a nice trick when using `Window.size`. This should work fine under the `Widget` + `Fbo` directly I think.